libs/execution/src/lib/util/mermaid-util.spec.ts

```diff
@@ -0,0 +1,161 @@
+      expect(getId()).toBe('b');
+      expect(getId()).toBe('c');
+      expect(getId()).toBe('d');
+      expect(getId()).toBe('e');
+      expect(getId()).toBe('f');
+      expect(getId()).toBe('g');
+      expect(getId()).toBe('h');
+      expect(getId()).toBe('i');
+      expect(getId()).toBe('j');
+      expect(getId()).toBe('k');
+      expect(getId()).toBe('l');
+      expect(getId()).toBe('m');
+      expect(getId()).toBe('n');
+      expect(getId()).toBe('o');
+      expect(getId()).toBe('p');
+      expect(getId()).toBe('q');
+      expect(getId()).toBe('r');
+      expect(getId()).toBe('s');
+      expect(getId()).toBe('t');
+      expect(getId()).toBe('u');
+      expect(getId()).toBe('v');
+      expect(getId()).toBe('w');
+      expect(getId()).toBe('x');
+      expect(getId()).toBe('y');
+      expect(getId()).toBe('z');
+      expect(getId()).toBe('ba');
+      expect(getId()).toBe('bb');
+    });
+  });
+  describe('Class Node', () => {
+    it('correctly prints a node', () => {
+      let node = new Node('nodeText', '[ ]');
+      expect(node.toString()).toBe(`${node.id}[nodeText]`);
+
+      node = new Node('otherNodeText', '( )');
+      expect(node.toString()).toBe(`${node.id}(otherNodeText)`);
+    });
+  });
+  describe('Class Edge', () => {
+    it('correctly prints an edge', () => {
+      const edge = new Edge('node', 'otherNode', 'edgeText', '-->');
+      expect(edge.toString()).toBe(`node ${edge.id}@-->|edgeText| otherNode`);
+    });
+  });
+  describe('Class EdgeAttribute', () => {
+    it('correctly prints an edge attribute', () => {
+      const edgeAttribute = new EdgeAttribute('edgeId');
+      expect(edgeAttribute.toString()).toBe('edgeId@{  }');
+
+      edgeAttribute.set('attr', 'val');
+      expect(edgeAttribute.toString()).toBe('edgeId@{ attr: val }');
+
+      edgeAttribute.set('attr2', 'val2');
+      expect(edgeAttribute.toString()).toBe(
+        'edgeId@{ attr: val, attr2: val2 }',
+      );
+    });
+  });
+  describe('Class ClassDefinition', () => {
+    it('correctly prints a class definition', () => {
+      const classDefinition = new ClassDefinition('className');
+      expect(classDefinition.toString()).toBe('classDef className');
+
+      classDefinition.set('attr', 'val');
+      expect(classDefinition.toString()).toBe('classDef className attr: val');
+
+      classDefinition.set('attr2', 'val2');
+      expect(classDefinition.toString()).toBe(
+        'classDef className attr: val,attr2: val2',
+      );
+    });
+  });
+  describe('Class Assignment', () => {
+    it('correctly prints a class assignment', () => {
+      const classAssignment = new ClassAssignment('id', 'className');
+      expect(classAssignment.toString()).toBe('class id className');
+    });
+  });
+  describe('Class Graph', () => {
+    it('correctly prints a Graph', () => {
+      const graph = new Graph('title');
+      expect(graph.toString()).toBe(`---
+title: title
+---
+flowchart TB
+`);
```

Comment:
For larger model we should use snapshot testing, but for now it's still fine like that.